<!--
  ~     Copyright 2017-2019 Bagaev Dmitry
  ~
  ~     Licensed under the Apache License, Version 2.0 (the "License");
  ~     you may not use this file except in compliance with the License.
  ~     You may obtain a copy of the License at
  ~
  ~         http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~     Unless required by applicable law or agreed to in writing, software
  ~     distributed under the License is distributed on an "AS IS" BASIS,
  ~     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~     See the License for the specific language governing permissions and
  ~     limitations under the License.
  -->

<filters-header headerTitle="CDR3" headerHelp="<p>Narrow search to a specific CDR3 motif or sequence different by no more than the specified number of mismatches from the query.</p>
<p>Sequence and pattern filter supports 20 amino acids, X for an unknown amino acid and amino acid groups specified with '[.]'. For example 'CASSL[ASR]PGATNEKLFF' will search
for 'CASSLaPGATNEKLFF', 'CASSLsPGATNEKLFF' and 'CASSLrPGATNEKLFF'. By default the search expects full match, which can be disabled by checking the substring checkbox.</p>
<p>Levenstein distance filter expects complete sequence with no wildcards, and user can specify the number of substitutions, insertions and deletions allowed during the
search.</p>">
</filters-header>
<div class="ui grid middle aligned">
    <div class="three wide column filters label"><b>Sequence or pattern</b></div>
    <div class="twelve wide column">
        <div class="ui right labeled fluid input" [ngClass]="{'error':!tcr.cdr3.isPatternValid()}">
            <input placeholder="Amino acid pattern, e.g. CSAR[DE]GTGNGYTF or CSARDXTGNGYTF."
                   type="text" [ngModel]="tcr.cdr3.pattern" (ngModelChange)="tcr.cdr3.checkPattern($event)">
            <div class="ui basic label">
                <div class="ui checkbox">
                    <input [(ngModel)]="tcr.cdr3.patternSubstring" type="checkbox">
                    <label>Substring</label>
                </div>
            </div>
        </div>
    </div>
    <div class="three wide column filters label"><b>CDR3 Length</b></div>
    <div class="eleven wide column">
        <slider [min]="tcr.cdr3.lengthMin" [max]="tcr.cdr3.lengthMax" [(model)]="tcr.cdr3.length"></slider>
    </div>
    <div class="three wide column filters label"> <b>By Levenstein distance</b></div>
    <div class="twelve wide column">
        <div class="ui fluid input" [ngClass]="{'error':!tcr.cdr3.isLevensteinValid()}">
            <input placeholder="Complete amino acid sequence, e.g. CASSLAPGATNEKLFF." [ngModel]="tcr.cdr3.levenstein" (ngModelChange)="tcr.cdr3.checkLevenstein($event)">
        </div>
    </div>
    <div class="three wide column"></div>
    <div class="four wide column">
        Substitutions
        <div class="ui fluid input">
            <input min="0" max="5" type="number" [ngModel]="tcr.cdr3.levensteinSubstitutions" (ngModelChange)="checkRangeInput('levensteinSubstitutions', $event, 0, 5)">
        </div>
    </div>
    <div class="four wide column">
        Insertions
        <div class="ui fluid input">
            <input min="0" max="3" type="number" [ngModel]="tcr.cdr3.levensteinInsertions" (ngModelChange)="checkRangeInput('levensteinInsertions', $event, 0, 3)">
        </div>
    </div>
    <div class="four wide column">
        Deletions
        <div class="ui fluid input">
            <input min="0" max="3" type="number" [ngModel]="tcr.cdr3.levensteinDeletions" (ngModelChange)="checkRangeInput('levensteinDeletions', $event, 0, 3)">
        </div>
    </div>
</div>